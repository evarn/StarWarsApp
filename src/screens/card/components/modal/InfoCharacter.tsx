import React from 'react';
import {StyleSheet, View} from 'react-native';

import Colors from '../../../../constants/colors';
import CircleCharacter from '../Circle';
import {getCheckFunction} from '../../../../features/getCheckFunction';
import Strings from './../../../../constants/strings';
import InfoCharcterTitleProps from './InfoCharcterTitleProps';

interface IInfoCharacter {
  hair_color: string | undefined;
  skin_color: string | undefined;
  height: string | undefined;
  mass: string | undefined;
}

const InfoCharacter = ({
  hair_color,
  skin_color,
  height,
  mass,
}: IInfoCharacter) => {
  return (
    <View style={styles.infoContainer}>
      <InfoCharcterTitleProps hair_color={hair_color} skin_color={skin_color} />
      <View style={styles.containerCircles}>
        {getCheckFunction(height) && (
          <View style={styles.containerCircle}>
            <CircleCharacter title={Strings.INFO_HEIGHT} param={height} />
          </View>
        )}
        {getCheckFunction(mass) && (
          <View style={styles.containerCircle}>
            <CircleCharacter title={Strings.INFO_MASS} param={mass} />
          </View>
        )}
      </View>
    </View>
  );
};

const styles = StyleSheet.create({
  infoContainer: {
    flex: 2,
    paddingHorizontal: 20,
    paddingVertical: 10,
    width: '100%',
    alignItems: 'center',
    backgroundColor: Colors.BLUE_2,
    borderBottomEndRadius: 20,
    borderBottomLeftRadius: 20,
    justifyContent: 'space-around',
  },
  containerCircle: {
    backgroundColor: Colors.WHITE_2,
    borderRadius: 8,
    marginHorizontal: 12,
  },
  containerCircles: {
    flexDirection: 'row',
  },
  containerProps: {
    backgroundColor: Colors.WHITE_2,
    margin: 8,
    borderRadius: 8,
    padding: 8,
    width: '100%',
  },
});

export default InfoCharacter;
