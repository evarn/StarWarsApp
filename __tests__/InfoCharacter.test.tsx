import 'react-native';
import React from 'react';
import {render, cleanup} from '@testing-library/react-native';
import InfoCharacter from '../src/screens/card/components/modal/InfoCharacter';

afterEach(cleanup);

describe('InfoCharacter', () => {
  it('InfoCharacter snapshoot', () => {
    const {toJSON} = render(
      <InfoCharacter
        hair_color={'blond'}
        skin_color={'blond'}
        eye_color={undefined}
        height={undefined}
        mass={undefined}
      />,
    );
    expect(toJSON()).toMatchSnapshot();
  });

  it('InfoCharacter text', () => {
    const {getByText} = render(
      <InfoCharacter
        hair_color={'blond'}
        skin_color={'blond'}
        eye_color={undefined}
        height={undefined}
        mass={undefined}
      />,
    );
    const hair_color = 'hair color: ';
    const foundTextHair = getByText(hair_color);

    const height = 'skin color: ';
    const foundTextHeight = getByText(height);

    expect(foundTextHair.props.children).toEqual(hair_color);
    expect(foundTextHeight.props.children).not.toEqual(hair_color);
  });
});
