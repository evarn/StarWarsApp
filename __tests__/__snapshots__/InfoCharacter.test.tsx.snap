// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InfoCharacter InfoCharacter snapshoot 1`] = `
<BVLinearGradient
  colors={
    [
      4280232547,
      4279697455,
    ]
  }
  endPoint={
    {
      "x": 0.5,
      "y": 1,
    }
  }
  locations={
    [
      0.42,
      1,
    ]
  }
  startPoint={
    {
      "x": 0.5,
      "y": 0,
    }
  }
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#17002F",
      "borderBottomLeftRadius": 16,
      "borderBottomRightRadius": 16,
      "flex": 3,
      "justifyContent": "space-around",
      "paddingHorizontal": 20,
      "paddingVertical": 10,
      "width": "100%",
    }
  }
>
  <View
    style={
      {
        "backgroundColor": "#FDFDFD",
        "borderRadius": 8,
        "margin": 8,
        "padding": 8,
        "width": "100%",
      }
    }
  >
    <Text
      style={
        {
          "color": "#000000",
          "fontFamily": "Karla",
          "fontSize": 16,
          "fontStyle": "normal",
          "fontWeight": "400",
          "lineHeight": 19,
        }
      }
    >
      <Text>
        hair color: 
      </Text>
      <Text>
        blond
      </Text>
    </Text>
    <Text
      style={
        {
          "color": "#000000",
          "fontFamily": "Karla",
          "fontSize": 16,
          "fontStyle": "normal",
          "fontWeight": "400",
          "lineHeight": 19,
        }
      }
    >
      <Text>
        skin color: 
      </Text>
      <Text>
        blond
      </Text>
    </Text>
  </View>
  <View
    style={
      {
        "flexDirection": "row",
      }
    }
  />
</BVLinearGradient>
`;
